### PR TITLE
refactor: add userinfo related stuff

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

- Reverts logto-io/swift#110
- Add new claims, resolves LOG-3993

## Test
- [x] sign in, check ID token claims ok
- [x] fetch userinfo, check claims ok
- [x] close the app and re-run, fetch userinfo, token refresh ok and check claims ok

### CODE REVIEW

File: Sources/Logto/Types/IdTokenClaims.swift [hunk 1]
```diff
@@ -7,17 +7,20 @@
...
     public let iat: Int64
     public let iss: String
 
-    // Scope `profile`
+    // Protocol
     public let name: String?
+    public let picture: String?
     public let username: String?
-    public let avatar: String?
-    public let roleNames: [String]?
```
Comment (charIeszhao) [inline Sources/Logto/Types/IdTokenClaims.swift:22]: We still have roleNames in idToken...

Comment (gao-sun) [reply Sources/Logto/Types/IdTokenClaims.swift:22]: it is useless now and shouldn't exist once RBAC implemented

Comment (charIeszhao) [reply Sources/Logto/Types/IdTokenClaims.swift:22]: OK, then we need to make this change in other SDKs as well

**Review by charIeszhao — APPROVED** [review]

File: Sources/Logto/Types/IdTokenClaims.swift (hunk 1 above)
Comment (gao-sun) [reply Sources/Logto/Types/IdTokenClaims.swift:22]: no need for now. when we implement RBAC we'll find the disharmony in SDK and remove it

**Review by charIeszhao — APPROVED** [review]

**Pull request merged**